### PR TITLE
Gemma2 in Nemo2 with Recipes

### PR DESCRIPTION
**Pull request by suiyoubi**
# What does this PR do ?

Add a one line overview of what this PR aims to accomplish.

**Collection**: [Note which collection this PR will affect]

# Changelog 
- Add specific line by line info of high level changes in this PR.

# Usage
* You can potentially add a usage example below

```python
# Add a code snippet demonstrating how to use this 
```

# GitHub Actions CI

The Jenkins CI system has been replaced by GitHub Actions self-hosted runners.

The GitHub Actions CI will run automatically when the "Run CICD" label is added to the PR.
To re-run CI remove and add the label again.
To run CI on an untrusted fork, a NeMo user with write access must first click "Approve and run".

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [ ] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

### CODE REVIEW

File: nemo/collections/llm/recipes/__init__.py [hunk 1]
```diff
@@ -90,7 +94,10 @@
     "nemotron4_22b_16k",
     "nemotron4_22b_64k",
     "nemotron4_340b",
-    "gpt3_175b",
```
Comment (cuichenx) [inline nemo/collections/llm/recipes/__init__.py:93]: Is this a merge error?

**Review by cuichenx — COMMENTED** [review]
Minor remark

File: nemo/collections/llm/recipes/__init__.py (hunk 1 above)
Comment (suiyoubi) [reply nemo/collections/llm/recipes/__init__.py:93]: good catch

**Review by cuichenx — APPROVED** [review]
LGTM!

File: nemo/collections/llm/gpt/model/gemma2.py [hunk 2]
```diff
@@ -0,0 +1,377 @@
...
+
+import torch
+from megatron.core import parallel_state
+from megatron.core.transformer.spec_utils import ModuleSpec
+from torch import nn
+
+from nemo.collections.llm.fn.activation import openai_gelu
+from nemo.collections.llm.gpt.model.base import GPTConfig, GPTModel
+from nemo.collections.llm.utils import Config
+from nemo.collections.nlp.models.language_modeling.megatron.gemma2.gemma2_spec import get_gemma2_layer_spec
```
Comment (cuichenx) [inline nemo/collections/llm/gpt/model/gemma2.py:27]: i think you have to make this import inside `gemma2_layer_spec`, otherwise the TE-free path will not work

**Review by cuichenx — APPROVED** [review]
